src/oc_erchef/schema/t/test_create_and_update_users.sql

```diff
@@ -0,0 +1,216 @@
+                                'aaab@chef.io',
+                                'this_is_not_a_key',
+                                0,
+                                'bbbb',
+                                'cccc',
+                                'bcrypt',
+                                'dddddddddddddddddddddddddddddddd',
+                                '2017-09-06 19:18:35',
+                                '2017-09-06 19:18:35',
+                                null,
+                                false,
+                                'foobarbaz',
+                                false),
+                       1,
+                       'run the add_user function v1');
+
+PREPARE "add id v1" AS SELECT public_key from users
+                         WHERE id='11111111111111111111111111111112';
+RETURN QUERY SELECT results_eq(
+    '"add id v1"',
+    ARRAY['this_is_not_a_key'],
+    'does not insert public_key into the users table for v1'
+);
+
+PREPARE "add key v1" AS SELECT public_key FROM keys
+                          WHERE id='11111111111111111111111111111112';
+RETURN QUERY SELECT is_empty(
+    '"add key v1"',
+    'does not insert public_key into the keys table for v1'
+);
+
+-- v0 update_user tests --
+
+PREPARE "before update email v0" AS SELECT email from users
+                                      WHERE id='11111111111111111111111111111111';
+RETURN QUERY SELECT results_eq(
+    '"before update email v0"',
+    ARRAY['aaaa@chef.io'],
+    'before update email address for v0'
+);
+
+RETURN QUERY SELECT is(update_user(0,
+                                   '--BEGIN PUBLIC KEY--END PUBLIC KEY--',
+                                   'bbbb',
+                                   'cccc',
+                                   'bcrypt',
+                                   'changed_something_in_my_name',
+                                   null,
+                                   false,
+                                   'changed_email_address_v0@chef.io',
+                                   'aaaa',
+                                   'dddddddddddddddddddddddddddddddd',
+                                   '2017-09-06 19:20:35',
+                                   false,
+                                   '11111111111111111111111111111111'),
+                       1,
+                       'run the update_user function v0');
+
+PREPARE "update id v0" AS SELECT public_key from users
+                            WHERE id='11111111111111111111111111111111';
+RETURN QUERY SELECT results_eq(
+    '"update id v0"',
+    ARRAY['this_is_not_a_key'],
+    'unchanged public_key in users table for v0'
+);
+
+PREPARE "after update email v0" AS SELECT email from users
+                                     WHERE id='11111111111111111111111111111111';
+RETURN QUERY SELECT results_eq(
+    '"after update email v0"',
+    ARRAY['changed_email_address_v0@chef.io'],
+    'after update email address for v0'
```

Comment:
I'd recommend stating these messages in terms of the desired behavior you're going after, instead of the mechanics of what is happening. That makes it easier to figure out what the overall intention of the code is.

Reply:
updated!

Reply:
done!